spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/outbound/PubSubMessageHandler.java

```diff
@@ -246,8 +247,8 @@ public void setHeaderMapper(HeaderMapper<Map<String, String>> headerMapper) {
 	}
 
 	@Override
-	protected void onInit() throws Exception {
-		super.onInit();
-		this.evaluationContext = ExpressionUtils.createStandardEvaluationContext(getBeanFactory());
+	public void setBeanFactory(BeanFactory beanFactory) {
+		super.setBeanFactory(beanFactory);
+		this.evaluationContext = ExpressionUtils.createStandardEvaluationContext(beanFactory);
```

Comment:
That' no again: I told in the previous review that we can't perform any `BeanFactory` access from the `setBeanFactory()`.
That's why I requested to reinstate the behavior in this class.

Why do you still leave this change here?
Any particular reason and why should not follow my advice?
And how may I determine the reason of this change if there is no explanations in the commit message?

Thanks for understanding.

Reply:
Should this line move to the `onInit`?

Reply:
Wasn't it there before? See the red code in this PR.

Reply:
Yes, I meant move *back*. :)

Reply:
@artembilan I think I understand now - you meant the only change  we need is to set bean factory in PubSubMessageChannelBinder.

Will do that.